wordpress-mobile/WordPress-Android
# Refactors NoteBlock to Kotlin

## Description
As an initial step towards the Milestone screen revamp this PR refactors some of the related code to Kotlin.

-----

## To Test:
1. Open the Notifications tab
2. Browse to different types of notifications
3. **Verify** no change in the functionality

-----

## Regression Notes

1. Potential unintended areas of impact

    - Notifications

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual Testing

5. What automated tests I added (or what prevented me from doing so)

    - I plan to add tests with a follow up PR

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
